### PR TITLE
fix-tabs

### PR DESCRIPTION
**Pull request by tomchang25**
# Description

Closes:  #3040
- Fixes bug where tabs selected attribute not working if manually change tab.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# Upcoming Release 
+# Upcoming Release
+
+## Bug Fixes:
+
+- Fixes bug where tabs selected attribute not working if manually change tab. Issuse #3040
```
Comment (freddyaboulton) [inline CHANGELOG.md:5]: Thanks @tomchang25 !

You need to add the github link, e.g. [PR 3055](https://github.com/gradio-app/gradio/pull/3055).

Do you mind moving this to the `## Bug Fixes` section below?

**Review by freddyaboulton — COMMENTED** [review]
Thanks so much @tomchang25 !

Was able to verify this fixes the issue 
tab_change_fix

Can you please fix the ui style check and the CHANGELOG? Happy to merge afterwards

File: CHANGELOG.md (hunk 1 above)
Comment (freddyaboulton) [reply CHANGELOG.md:5]: Also, can you revert all the other changes to this file except adding the line for this PR?

File: ui/pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.3
```
[the remark this answers is not in the document]
Comment (pngwn) [reply ui/pnpm-lock.yaml:1]: They are compatible so i shouldn't matter much.

**Review by freddyaboulton — APPROVED** [review]
Thank you so much @tomchang25 !

**Pull request merged**